src/components/environments-summary/environment-ingress.js

```diff
@@ -1,40 +1,98 @@
 import React from 'react';
 import { faLink } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons';
 
 import usePollComponents from './use-poll-components';
 
 const URL_VAR_NAME = 'RADIX_PUBLIC_DOMAIN_NAME';
-const MAX_DISPLAY_NR_COMPONENT = 3;
+const MAX_DISPLAY_NR_COMPONENT = 4;
+
+const outdatedOrFailedComponent = (component) => {
+  if (component.status === 'Outdated') {
+    return (
+      <span
+        class="env-summary-image-outdated-warning"
+        title="Component is running an outdated image"
+      >
+        <FontAwesomeIcon icon={faExclamationCircle} />
+      </span>
+    );
+  }
+  if (component.status === 'Failing') {
+    return (
+      <span
+        class="env-summary-image-outdated-warning"
+        title="Component is failing"
+      >
+        <FontAwesomeIcon icon={faExclamationCircle} />
+      </span>
+    );
+  }
+};
 
 const EnvironmentIngress = ({ appName, deploymentName }) => {
   const [componentsPollState] = usePollComponents(appName, deploymentName);
+
   let components = [];
   if (componentsPollState && componentsPollState.data) {
-    components = componentsPollState.data.filter(
-      (component) => component.variables[URL_VAR_NAME]
+    components = componentsPollState.data;
+  }
+
+  let publicComponents = [];
+  let passiveComponents = [];
+  if (components) {
+    publicComponents = components.filter(
+      (comp) => comp.variables[URL_VAR_NAME]
+    );
+    passiveComponents = components.filter(
+      (comp) => !comp.variables[URL_VAR_NAME]
     );
   }
 
   if (components.length <= 0) {
     return <div />;
   }
-  const tooManyPublicComponents = components.length > MAX_DISPLAY_NR_COMPONENT;
+  const tooManyPublicComponents =
+    publicComponents.length > MAX_DISPLAY_NR_COMPONENT;
+
+  const tooManyPassiveComponents =
+    passiveComponents.length > MAX_DISPLAY_NR_COMPONENT;
+
   if (tooManyPublicComponents) {
-    components = components.slice(0, MAX_DISPLAY_NR_COMPONENT);
+    publicComponents = publicComponents.slice(0, MAX_DISPLAY_NR_COMPONENT);
+  }
+
+  if (tooManyPassiveComponents) {
+    passiveComponents = passiveComponents.slice(0, MAX_DISPLAY_NR_COMPONENT);
   }
```

Comment:
Would it be better to use one slice for borth public and passive components by appending the passive (only those failing and outdated) to the public slice.
Currently, if there are four public components and some passive components that are failing, the height of the comp-list would exceed the environment box.
Using one slice also reduces code duplication.

Reply:
As discussed, combining the slices could result in none of the failing components to show in the list. Reduced number of max components to account for layout issues.